tests/unit/components/Globe.spec.ts

```diff
@@ -0,0 +1,228 @@
+
+      appStore.currentScenario.parameters = { country: "NOR" };
+
+      await component.vm.$nextTick();
+
+      await waitFor(() => {
+        expect(chart.get("rotationX")).toBeCloseTo(-12, 0);
+        expect(chart.get("rotationY")).toBeCloseTo(-39, 0);
+        expect(zoomToGeoBoundsSpy).toHaveBeenCalled();
+      }, { timeout: 2500 /* >= rotateDuration */ });
+    });
+
+    it("zooms back out and re-starts the gentle rotation animation when the user navigates back to the new scenario page", async () => {
+      const testPinia = mockPinia({
+        metadata: {
+          ...mockedMetadata,
+          parameters: [...selectParameters, { ...globeParameter, id: "country" }, updatableNumericParameter],
+        },
+      });
+      const appStore = useAppStore(testPinia);
+
+      const component = await mountSuspended(Globe, {
+        global: { plugins: [testPinia] },
+      });
+
+      const chart = (component.vm as any).setUpChart();
+
+      const zoomToGeoBoundsSpy = vi.spyOn(chart, "zoomToGeoBounds");
+      const goHomeSpy = vi.spyOn(chart, "goHome");
+
+      appStore.currentScenario.parameters = { country: "NOR" };
+
+      await component.vm.$nextTick();
+
+      await waitFor(() => {
+        expect(chart.get("rotationX")).toBeCloseTo(-12, 0);
+        expect(chart.get("rotationY")).toBeCloseTo(-39, 0);
+        expect(zoomToGeoBoundsSpy).toHaveBeenCalled();
+      }, { timeout: 2500 /* >= rotateDuration */ });
+
+      // Simulate the changes to app store that are performed by the 'scenarios/new' page's onMounted hook
+      appStore.globe.interactive = false;
+      appStore.globe.highlightedCountry = null;
+
+      await component.vm.$nextTick();
+
+      expect(component.vm.gentleRotateAnimation.stopped).toBe(false);
+      await waitFor(() => {
+        expect(goHomeSpy).toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe("on the results page", () => {
+    beforeEach(() => {
+      mockRoute.mockReturnValue({
+        path: "/scenarios/1",
+      });
+    });
+
+    it("updating highlightedCountry in store (as when a country is selected from the drop-down) should not trigger a rotation to that country", async () => {
+      const testPinia = mockPinia();
+      const appStore = useAppStore(testPinia);
+      const component = await mountSuspended(Globe, {
+        global: { plugins: [testPinia] },
+      });
+
+      const chart = (component.vm as any).setUpChart();
+
+      appStore.globe.highlightedCountry = "GBR";
+
+      await component.vm.$nextTick();
+
+      const gbrSeries = chart.series._values[8];
+      const originalColor = gbrSeries._settings.fill;
+      const originalX = chart.get("rotationX");
+      const originalY = chart.get("rotationY");
+
+      await waitFor(() => {
+        expect(gbrSeries._settings.fill).not.toBe(originalColor);
+        expect(chart.get("rotationX")).toBe(originalX);
+        expect(chart.get("rotationY")).toBe(originalY);
+      });
```

Comment:
So it *should* update the colour but shouldn't update the rotation. And by the time it updates the colour it would have updated the rotation if it was going to? So doing this as a waitFor is ok..?